src/vector/geometry.rs

```diff
@@ -197,3 +217,11 @@ impl Drop for Geometry {
         }
     }
 }
+
+impl Clone for Geometry {
+    fn clone(&self) -> Geometry {
+        let c_geometry = self.c_geometry_ref.borrow();
+        let new_c_geom = unsafe { ogr::OGR_G_Clone(c_geometry.unwrap())};
```

Comment:
I think you are right that `clone` can not return a `Result`.  @frewsxcv can you help here?